### PR TITLE
[cxx] Put extern "C" in MONO_LLVM_INTERNAL.

### PR DESCRIPTION
**Pull request by jaykrell**
Extracted from https://github.com/mono/mono/pull/17325.
This might give Windows a chance at having loaded LLVM also, if desired (probably not).

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
/cc @lambamdageek

This is so more lanes can pass CI as C++.

File: mono/utils/mono-dl.h [hunk 1]
```diff
@@ -29,11 +29,11 @@ typedef struct {
...
 
 
-MONO_API MonoDl*     mono_dl_open       (const char *name, int flags, char **error_msg) MONO_LLVM_INTERNAL;
+MONO_API MonoDl* mono_dl_open  (const char *name, int flags, char **error_msg) MONO_LLVM_INTERNAL_NO_EXTERN_C;
 MONO_EXTERN_C
-char*       mono_dl_symbol     (MonoDl *module, const char *name, void **symbol) MONO_LLVM_INTERNAL;
+char*       mono_dl_symbol     (MonoDl *module, const char *name, void **symbol) MONO_LLVM_INTERNAL_NO_EXTERN_C;
 MONO_EXTERN_C
-void        mono_dl_close      (MonoDl *module) MONO_LLVM_INTERNAL;
+void        mono_dl_close      (MonoDl *module) MONO_LLVM_INTERNAL_NO_EXTERN_C;
```
Comment (lambdageek) [inline mono/utils/mono-dl.h]: These are surprising.

Comment (lambdageek) [reply mono/utils/mono-dl.h]: Oh wait, no they're not, I guess. It's because we want them `extern "C"` always, even if we're not doing loaded LLVM.

Comment (jaykrell) [reply mono/utils/mono-dl.h]: Right. It is a delicate dance among related but not contradictory factors:

 - functions
 - data
 -MONO_API
 - LOADED_LLVM being conditional. These could also be moved to start of line I think, and maybe mildly preferable, for consistency, and to give LOADED_LLVM on Windows a chance, if that ever matters.

**Pull request merged**